clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -223,8 +232,65 @@ public void testFencingWhenStateIsReconciling() {
         verify(subscriptionState).assignFromSubscribed(Collections.emptySet());
     }
 
+    @Test
+    public void testFencingWhenStateIsPrepareLeaving() {
+        MembershipManagerImpl membershipManager = createMemberInStableState();
+
+        // Start leaving group, blocked waiting for commit of all consumed to complete.
+        CompletableFuture<Void>  commitResult = mockPrepareLeavingStuckCommitting();
+        membershipManager.leaveGroup();
+        assertEquals(MemberState.PREPARE_LEAVING, membershipManager.state());
+
+        // Get fenced while preparing to leave the group. Member should not try to rejoin and
+        // continue leaving the group as it was before getting fenced.
+        mockMemberHasAutoAssignedPartition();
+        membershipManager.transitionToFenced();
```

Comment:
Is there a way to verify that we don't (re-)trigger the lost callback here?

Reply:
Yes, using the `MockRebalanceListener`. Done.